### ISSUE TITLE
Display processor properties

### ISSUE THREAD

**Issue by dougcooper** [palindromicity/dotifi#67]
I would like to be able to optionally print the properties and their associated values.

**Comment by ottobackwards** [comment]
So, inside of the 'box' for a component?  That may make each box rather large but I can see how that would be useful

**Comment by dougcooper** [comment]
Exactly. I was looking for a way to easily document the flows for presentation purposes outside of nifi and this tool is perfect. But ideally it would capture the customization of the processor as well. Not sure how the dot templating stuff works but I wonder if it would be possible to specify which properties we want to show?

**Comment by ottobackwards** [comment]
I like to think of the templates as strictly for the visualization, not the content.  What I would be thinking would be to:
1st -> allow a 'switch' in the configuration yaml to output properties
2nd ( I need to think of how yucky the configuration would get ) all for per processor type property lists

Also, this makes me think that either dotifi or another peer tool to document a flow outputting markdown, or some combination of markdown and the dot file / image would be cool.

So imagine if dotifi output a web page and the dot file, such that it had an image of the flow, and beneath that a listing of the processors and properties

**Comment by ottobackwards** [comment]
I think the 'how complicated the configuration gets' isn't much of an issue.  I've always imagined that someone would have some number of configurations, for different use cases, which were one time writes.

**Comment by dougcooper** [comment]
I really like that idea. Like each processor could have an href that links to the details for that processor. That would be perfect.

**Comment by ottobackwards** [comment]
I could use a specified HTML template to configure how that comes out too

**Comment by ottobackwards** [comment]
People should be able to pipeline that into whatever they want

**Comment by dougcooper** [comment]
Pandoc is great for pipelining....

**Comment by ottobackwards** [comment]
so, would html be the best output or markdown I wonder?

**Comment by dougcooper** [comment]
I think markdown. You can give pandoc a latex template and it will handle all the formatting. https://pandoc.org/MANUAL.html#templates

**Comment by ottobackwards** [comment]
Would that handle markdown with embedded raw html tags?  I don't think I can do everything I want to do with raw markdown....

**Comment by dougcooper** [comment]
Generally speaking markdown allows embedded html. Pandoc supports some extensions as well. https://pandoc.org/MANUAL.html#markdown-variants.
